src/backend/distributed/commands/index.c

```diff
@@ -442,23 +444,6 @@ PostprocessIndexStmt(Node *node, const char *queryString)
 	CommitTransactionCommand();
 	StartTransactionCommand();
 
-	/* now, update index's validity in a way that can roll back */
-	Relation pg_index = table_open(IndexRelationId, RowExclusiveLock);
-
-	HeapTuple indexTuple = SearchSysCacheCopy1(INDEXRELID, ObjectIdGetDatum(
-												   indexRelationId));
-	Assert(HeapTupleIsValid(indexTuple)); /* better be present, we have lock! */
-
-	/* mark as valid, save, and update pg_index indexes */
-	Form_pg_index indexForm = (Form_pg_index) GETSTRUCT(indexTuple);
-	indexForm->indisvalid = true;
```

Comment:
This logic used to mark the index as valid without committing, such that a failure during distributed DDL execution would roll back to the invalid state set above. However, after adding the CommitTransactionCommand in ExecuteDistributedDDLJob we would always commit the valid state. I therefore set the valid state after ExecuteDistributedDDLJob completes, in FinishConcurrentIndexCommand.